File: functions/ffmpeg/variants/go/0_flac_get_max_frame_size_2.go
package main

func flacGetMaxFrameSize(blocksize, ch, bps int32) int32 {
    header := int32(16)
    channels := (7 + bps + 7) / 8
    count := header + ch*channels
    var frameData int32
    if ch == 2 {
        frameData = ((2*bps + 1) * blocksize + 7) / 8
    } else {
        frameData = (ch * bps * blocksize + 7) / 8
    }
    count += frameData + 2
    return count
}


func main(){
    flacGetMaxFrameSize(0,0,0)
}

File: functions/ffmpeg/variants/c/0_flac_get_max_frame_size_2.c
#include <stdint.h>

static int flac_get_max_frame_size(int blocksize, int ch, int bps) 
{
    int base = 16;
    int audio_frame_size = (ch * (bps + 14)) / 8;
    int side_info_size = (ch == 2) ? ((2 * bps + 1) * blocksize + 7) / 8 : (ch * bps * blocksize + 7) / 8;
    
    return base + audio_frame_size + side_info_size + 2;
}


int main(){
    flac_get_max_frame_size(0,0,0);
}
